From 389814c78216f8f969f99bd87d58d7f386e0301a Mon Sep 17 00:00:00 2001
From: Antoine Martin (ayakael) <dev@ayakael.net>
Date: Fri, 5 Aug 2022 15:05:30 +0000
Subject: [PATCH 1/1] improved VERSION_ID parsing

Runtime doesn't generate a proper distro-rid for Alpine, leaving the
last version digit. Patch fixes this by adding new VERSION_ID parsing
logics.

---
 eng/native/init-distro-rid.sh | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/src/runtime/eng/native/init-distro-rid.sh b/src/runtime/eng/native/init-distro-rid.sh
index f71aa8640b1..fce4092a43d 100644
--- a/src/runtime/eng/native/init-distro-rid.sh
+++ b/src/runtime/eng/native/init-distro-rid.sh
@@ -41,16 +41,21 @@ initNonPortableDistroRid()
             # We have forced __PortableBuild=0. This is because -portablebuld
             # has been passed as false.
             if (( isPortable == 0 )); then
-                if [[ "${ID}" == "rhel" || "${ID}" == "rocky" || "${ID}" == "alpine" ]]; then
-                    # remove the last version digit
-                    VERSION_ID="${VERSION_ID%.*}"
-                fi
-
                 if [ -z "${VERSION_ID+x}" ]; then
                         # Rolling release distros do not set VERSION_ID, so omit
                         # it here to be consistent with everything else.
                         nonPortableBuildID="${ID}-${buildArch}"
                 else
+                        local VERSION_ID_DOT="${VERSION_ID//[^.]}"
+                        while [[ ${#VERSION_ID_DOT} -gt 1 ]]; do
+                                VERSION_ID="${VERSION_ID%.*}"
+                                local VERSION_ID_DOT="${VERSION_ID//[^.]}"
+                        done
+                        local VERSION_ID_DASH="${VERSION_ID//[^_]}"
+                        while [[ ${#VERSION_ID_DASH} -ge 1 ]]; do
+                                VERSION_ID="${VERSION_ID%_*}"
+                                local VERSION_ID_DASH="${VERSION_ID//[^_]}"
+                        done
                         nonPortableBuildID="${ID}.${VERSION_ID}-${buildArch}"
                 fi
             fi
-- 
2.37.1
